File: Roomies/roomies/src/service/PropertyService.js
import axios from "axios";



const Prop_Get_By_iD_BASE_URL = "http://localhost:8080/property"


class PropertyService {

    getById(id){
        return axios.get(Prop_Get_By_iD_BASE_URL+"/"+id);
    }

    getId=(flat_no,society_name)=>{
        return axios.get(Prop_Get_By_iD_BASE_URL+"/"+flat_no+"/"+society_name)
    }
}

export default PropertyService;

File: RoomieProject/src/main/java/com/demo/service/PropertyService.java
package com.demo.service;

import java.util.List;
import java.util.Optional;

import com.demo.beans.Property;

public interface PropertyService {

	Property addProperty(Property property);

	List<Property> getAll();

	Optional<Property> getById(int id);

	Property update(Property property);

	void deleteProperty(int id);

	Property searchByCity(String city);

	List<String> searchCity();

	Property getId(int flat_no, String society_name);

}
